internal/gps/result.go

```diff
@@ -65,37 +66,91 @@ func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool, logger *log
 		return err
 	}
 
-	var wg sync.WaitGroup
-	errCh := make(chan error, len(l.Projects()))
+	lps := l.Projects()
+
+	type resp struct {
+		i   int
+		err error
+	}
+	respCh := make(chan resp, len(lps))
+	writeCh := make(chan int, len(lps))
+	cancel := make(chan struct{})
 
-	for _, p := range l.Projects() {
-		wg.Add(1)
-		go func(p LockedProject) {
+	// Queue work.
+	for i := range lps {
+		writeCh <- i
+	}
+	close(writeCh)
+	// Launch writers.
+	writers := runtime.GOMAXPROCS(-1)
```

Comment:
Perhaps an optimal number is some factor of `GOMAXPROCS`, or ultimately something configurable.
Related discussion on how this might be configurable: https://github.com/golang/dep/pull/1028#issuecomment-324902148

Reply:
i'm not sure that GOMAXPROCS is the right thing, here - the goroutines themselves aren't actually doing work, but end up mostly in a sleep mode waiting for their spawned subprocesses. limiting it to GOMAXPROCS seems likely to be lower than we want.

i think i'd rather just pick an arbitrary number to begin with - let's say 16 - and see how we do.